instrumentation-api-incubator/src/main/jflex/SqlSanitizer.jflex

```diff
@@ -52,6 +54,9 @@ WHITESPACE          = [ \t\r\n]+
   // max length of the sanitized statement - SQLs longer than this will be trimmed
   static final int LIMIT = 32 * 1024;
 
+  private static final Pattern IN_STATEMENT_PATTERN = Pattern.compile("\\sin\\s*\\(\\s*\\?[\\s?,]*?\\)", Pattern.CASE_INSENSITIVE);
```

Comment:
I think this also matches inputs like ` in (?,,,???)` perhaps using `"(\\sin\\s*)\\(\\s*\\?\\s*(,\\s*\\?\\s*)*\\)"` and replacing with `"$1(?)"` would be better. These regular expressions are hard to parse, maybe we should try to document them to make them easier to understand?

Reply:
> Unfortunately the (,\\s*\\?\\s*)* part of that pattern causes a stack overflow for IN statements with many values

using a possessive quantifier should fix this, try `"(\\sin\\s*)\\(\\s*\\?\\s*(,\\s*\\?\\s*)*+\\)"`

Reply:
thanks, that solves the problem